### ISSUE TITLE
[Accessibility] Screen reader does not announce InfoBar when toggled in Navigated event handler

### ISSUE THREAD

**Issue by miksalmon** [microsoft/microsoft-ui-xaml#10733]
### Describe the bug

There is an issue with the narrator announcement of an InfoBar when opened or closed in a Navigated event. 

Narrator announcement work correctly when opened or closed on its own. 

However, when navigating which triggers the InfoBar to be IsOpen = true, there is no screen reader announcement. NarratorBuddy shows it is being announced, but it cannot be heard. It seems to be overtaken by the NavigationView's selected menu item announcement.

When navigating where it closes the InfoBar, no announcement is made at all saying it is dismissed.

Here is a min repro app: https://github.com/miksalmon/InfoBarNarratorIssue

### Steps to reproduce the bug

1. Navigate to Page 1
2. Notice the InfoBar does not get announced
3. Navigate to Page 2
4. Notice the InfoBar dismissed does not get announced

### Actual behavior

1. InfoBar does not get announced when IsOpen is set to true (in a Navigated event handler)
2. InfoBar dismissed does not get announced when IsOpen is set to true (in a Navigated event handler)

### Expected behavior

InfoBar should be announced when IsOpen is set to true.
InfoBar dismissed should be announced when IsOpen is set to false.

This behaviour can be seen using the checkbox in the min repro app. When not tied to a Navigated event, it works correctly.

### NuGet package version

None

**Comment by miksalmon** [comment]
My bad. Just made it public.

Loaded wouldn't work because in my use case, the InfoBar is a level above the pages (global) and I rely on the navigation to turn it on or off. 

I will try the Dispatch and after CompositionTargerRender. Otherwise, I might refactor the infobar to be in the pages and try with loaded instead. 

Thanks!

**Comment by michael-hawker** [comment]
Thanks! Yeah, I'm wondering if this is a timing thing as all the page content is getting loaded at the same time the InfoBar is opening.

So, it may just be getting interrupted and overwritten vs. queuing or something. At least if you play around with the timing in different phase of XAML loading it could maybe help determine more of what's happening; so keep us posted if you discover anything!

**Comment by miksalmon** [comment]
Dispatching on the UI thread worked, most likely because it resolved the timing issue to ensure it is queued properly as opposed to competing with the NavigationView menu item selection announcement.
